lua/spectre_common.lua

```diff
@@ -105,25 +127,39 @@ local function determine_if_cacheable(url, namespace, request_headers)
                 refresh_cache = false,
             }
 
-            -- Only cache GET and HEAD requests
-            local http_method = ngx.req.get_method()
-            if http_method ~= 'GET' and http_method ~= 'HEAD' then
-                cacheability_info.is_cacheable = false
-                cacheability_info.reason = 'non-cacheable-method'
-            end
-
             -- Check if client sent no-cache header
             if has_marker_headers(request_headers, PLEASE_DO_NOT_CACHE_HEADERS) then
                 cacheability_info.is_cacheable = false
                 cacheability_info.reason = 'no-cache-header'
                 cacheability_info.refresh_cache = true
             end
-            break
```

Comment:
Don't we still need the `break` to exit the loop once we've found a matching entry? It will still work without it, but it'll be less efficient

Reply:
Fixed